packages/design-system/bin/transform-figma-tokens.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-console */
+
+import { execSync } from "child_process";
+import { readFileSync, writeFileSync, existsSync, rmSync } from "fs";
+
+const SOURCE_FILE = "./src/__generated__/figma-design-tokens.json";
+const TMP_OUTPUT_FILE = "./src/__generated__/figma-design-tokens.tmp";
+const OUTPUT_FILE = "./src/__generated__/figma-design-tokens.ts";
+
+const traverse = (
+  node: unknown,
+  nodePath: string[],
+  fn: (path: string[], type: string, value: unknown) => void
+) => {
+  if (typeof node !== "object" || node === null) {
+    return;
+  }
+
+  const entries = Object.entries(node);
+
+  // TypeScript forces us to use entries to read properties of an unknown object
+  const type = entries.find(([key]) => key === "type");
+  const value = entries.find(([key]) => key === "value");
+
+  // if there's a `type` and `value` properties, treat it as a tree leaf
+  if (type && value && typeof type[1] === "string") {
+    fn(nodePath, type[1], value[1]);
+    // otherwise, traverse deeper
+  } else {
+    for (const [key, value] of entries) {
+      traverse(value, [...nodePath, key], fn);
+    }
+  }
+};
+
+const toCamelCase = (text: string) =>
+  text
+    .split(/[^a-z0-9]+/i)
+    .map((word) => {
+      if (word.length === 0) {
+        return "";
+      }
+      return word[0].toUpperCase() + word.slice(1);
+    })
+    .join("");
+
+const trimPrefix = (prefix: string, text: string) => {
+  if (text.toLocaleLowerCase().startsWith(prefix.toLocaleLowerCase())) {
+    return text.slice(prefix.length);
+  }
+  return text;
+};
+
+const deCapitalize = (text: string) => text[0].toLowerCase() + text.slice(1);
+
+const pathToName = (path: string[], type: string) =>
+  deCapitalize(trimPrefix(type, toCamelCase(path.join("-"))));
+
+const main = () => {
+  execSync(`token-transformer ${SOURCE_FILE} ${TMP_OUTPUT_FILE}`, {
```

Comment:
You can try to use api directly. The code may be a little simpler with it
https://github.com/tokens-studio/figma-plugin/blob/main/token-transformer/index.js#L7

Reply:
Yeah, but this is not official public API, so may stop working at any time